vendor/zkllvm-metacraft-circuits/src/circuit_utils/circuit_byte_utils.h

```diff
@@ -108,8 +104,8 @@ namespace circuit_byte_utils {
 
         T result = 0;
         if constexpr (LittleEndian) {
-            for (unsigned int i = sizeof(T); i > 0; i--) {
-                result = (result << 8) + *(first_element + i - 1);
+            for (unsigned i = 0; i < sizeof(T); ++i) {
+                result = (result << 8) + *(first_element + sizeof(T) - i - 1);
```

Comment:
`first_element`and `last_element` could be `begin` and `end` for consistency. Also they are not elements